### ISSUE TITLE
coda and segno encoding

### ISSUE THREAD

**Issue by craigsapp** [music-encoding/music-encoding#520]
How are coda and segno encoded?  They should use `<dir>` as mentioned in the guidelines:

http://music-encoding.org/docs/MEI_Guidelines_v3.0.0.pdf

But there is not an example, which would better than me using telepathy to figure it out :-)

Example to fill in:

<img width="680" alt="screen shot 2018-05-11 at 10 45 17 pm" src="https://user-images.githubusercontent.com/3487289/39954046-cc06f48c-556c-11e8-8435-a28233baea45.png">

MEI data without coda or segno:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/4.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="4.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt />
        </fileDesc>
        <encodingDesc>
            <appInfo>
                <application isodate="2018-05-11T22:41:13" version="2.0.0-dev-509c169-dirty">
                    <name>Verovio</name>
                    <p>Transcoded from Humdrum</p>
                </application>
            </appInfo>
        </encodingDesc>
        <workDesc>
            <work>
                <titleStmt>
                    <title />
                </titleStmt>
            </work>
        </workDesc>
    </meiHead>
    <music>
        <body>
            <mdiv xml:id="mdiv-0000000775059617">
                <score xml:id="score-0000001883574307">
                    <scoreDef xml:id="scoredef-0000000355467483" midi.bpm="400">
                        <staffGrp xml:id="staffgrp-0000001812563648">
                            <staffDef xml:id="staffdef-0000001748516128" clef.shape="G" clef.line="2" key.sig="0" meter.count="4" meter.unit="4" n="1" lines="5">
                                <label xml:id="label-0000001930042499" />
                            </staffDef>
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-0000000562905122">
                        <measure xml:id="measure-L5" n="1">
                            <staff xml:id="staff-L5F1N1" n="1">
                                <layer xml:id="layer-L5F1N1" n="1">
                                    <note xml:id="note-L6F1" dur="1" oct="4" pname="g" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L7" n="2">
                            <staff xml:id="staff-L7F1N1" n="1">
                                <layer xml:id="layer-L7F1N1" n="1">
                                    <note xml:id="note-L8F1" dur="1" oct="4" pname="d" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L9" n="3">
                            <staff xml:id="staff-L9F1N1" n="1">
                                <layer xml:id="layer-L9F1N1" n="1">
                                    <note xml:id="note-L10F1" dur="1" oct="4" pname="g" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L11" n="4">
                            <staff xml:id="staff-L11F1N1" n="1">
                                <layer xml:id="layer-L11F1N1" n="1">
                                    <note xml:id="note-L12F1" dur="1" oct="4" pname="f" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L13" n="5">
                            <staff xml:id="staff-L13F1N1" n="1">
                                <layer xml:id="layer-L13F1N1" n="1">
                                    <note xml:id="note-L14F1" dur="1" oct="4" pname="g" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L15" n="6">
                            <staff xml:id="staff-L15F1N1" n="1">
                                <layer xml:id="layer-L15F1N1" n="1">
                                    <note xml:id="note-L16F1" dur="1" oct="4" pname="g" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L18" n="7">
                            <staff xml:id="staff-L18F1N1" n="1">
                                <layer xml:id="layer-L18F1N1" n="1">
                                    <note xml:id="note-L19F1" dur="1" oct="4" pname="a" accid.ges="n" />
                                </layer>
                            </staff>
                            <dir xml:id="dir-L19F1" place="above" staff="1" tstamp="1.000000">
                                <rend xml:id="rend-0000002013122821" fontstyle="normal">D.S.</rend>
                            </dir>
                        </measure>
                        <measure xml:id="measure-L20" n="8">
                            <staff xml:id="staff-L20F1N1" n="1">
                                <layer xml:id="layer-L20F1N1" n="1">
                                    <note xml:id="note-L21F1" dur="1" oct="4" pname="g" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L22" n="9">
                            <staff xml:id="staff-L22F1N1" n="1">
                                <layer xml:id="layer-L22F1N1" n="1">
                                    <note xml:id="note-L23F1" dur="1" oct="4" pname="e" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L24" n="10">
                            <staff xml:id="staff-L24F1N1" n="1">
                                <layer xml:id="layer-L24F1N1" n="1">
                                    <note xml:id="note-L25F1" dur="1" oct="5" pname="e" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L26" n="11">
                            <staff xml:id="staff-L26F1N1" n="1">
                                <layer xml:id="layer-L26F1N1" n="1">
                                    <note xml:id="note-L27F1" dur="1" oct="5" pname="c" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L28" n="12">
                            <staff xml:id="staff-L28F1N1" n="1">
                                <layer xml:id="layer-L28F1N1" n="1">
                                    <note xml:id="note-L29F1" dur="1" oct="4" pname="a" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-L30" right="end" n="13">
                            <staff xml:id="staff-L30F1N1" n="1">
                                <layer xml:id="layer-L30F1N1" n="1">
                                    <note xml:id="note-L31F1" dur="1" oct="4" pname="g" accid.ges="n" />
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

Source MusicXML data:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE score-partwise PUBLIC "-//Recordare//DTD MusicXML 3.0 Partwise//EN" "http://www.musicxml.org/dtds/partwise.dtd">
<score-partwise>
  <work>
    <work-title>Title</work-title>
    </work>
  <identification>
    <creator type="composer">Composer</creator>
    <encoding>
      <software>MuseScore 2.1.0</software>
      <encoding-date>2018-05-11</encoding-date>
      <supports element="accidental" type="yes"/>
      <supports element="beam" type="yes"/>
      <supports element="print" attribute="new-page" type="yes" value="yes"/>
      <supports element="print" attribute="new-system" type="yes" value="yes"/>
      <supports element="stem" type="yes"/>
      </encoding>
    </identification>
  <defaults>
    <scaling>
      <millimeters>7.05556</millimeters>
      <tenths>40</tenths>
      </scaling>
    <page-layout>
      <page-height>1683.36</page-height>
      <page-width>1190.88</page-width>
      <page-margins type="even">
        <left-margin>56.6929</left-margin>
        <right-margin>56.6929</right-margin>
        <top-margin>56.6929</top-margin>
        <bottom-margin>113.386</bottom-margin>
        </page-margins>
      <page-margins type="odd">
        <left-margin>56.6929</left-margin>
        <right-margin>56.6929</right-margin>
        <top-margin>56.6929</top-margin>
        <bottom-margin>113.386</bottom-margin>
        </page-margins>
      </page-layout>
    <word-font font-family="FreeSerif" font-size="10"/>
    <lyric-font font-family="FreeSerif" font-size="11"/>
    </defaults>
  <credit page="1">
    <credit-words default-x="595.44" default-y="1626.67" justify="center" valign="top" font-size="24">Title</credit-words>
    </credit>
  <credit page="1">
    <credit-words default-x="1134.19" default-y="1526.67" justify="right" valign="bottom" font-size="12">Composer</credit-words>
    </credit>
  <part-list>
    <score-part id="P1">
      <part-name>Piano</part-name>
      <part-abbreviation>Pno.</part-abbreviation>
      <score-instrument id="P1-I1">
        <instrument-name>Piano</instrument-name>
        </score-instrument>
      <midi-device id="P1-I1" port="1"></midi-device>
      <midi-instrument id="P1-I1">
        <midi-channel>1</midi-channel>
        <midi-program>1</midi-program>
        <volume>78.7402</volume>
        <pan>0</pan>
        </midi-instrument>
      </score-part>
    </part-list>
  <part id="P1">
    <measure number="1" width="312.94">
      <print>
        <system-layout>
          <system-margins>
            <left-margin>0.00</left-margin>
            <right-margin>-0.00</right-margin>
            </system-margins>
          <top-system-distance>170.00</top-system-distance>
          </system-layout>
        </print>
      <attributes>
        <divisions>1</divisions>
        <key>
          <fifths>0</fifths>
          </key>
        <time>
          <beats>4</beats>
          <beat-type>4</beat-type>
          </time>
        <clef>
          <sign>G</sign>
          <line>2</line>
          </clef>
        </attributes>
      <note default-x="75.17" default-y="-30.00">
        <pitch>
          <step>G</step>
          <octave>4</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        </note>
      </measure>
    <measure number="2" width="254.85">
      <direction placement="above">
        <direction-type>
          <segno/>
          </direction-type>
        <sound segno="segno"/>
        </direction>
      <note default-x="12.00" default-y="-45.00">
        <pitch>
          <step>D</step>
          <octave>4</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        </note>
      </measure>
    <measure number="3" width="254.85">
      <note default-x="12.00" default-y="-30.00">
        <pitch>
          <step>G</step>
          <octave>4</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        </note>
      </measure>
    <measure number="4" width="254.85">
      <note default-x="12.00" default-y="-35.00">
        <pitch>
          <step>F</step>
          <octave>4</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        </note>
      </measure>
    <measure number="5" width="293.37">
      <print new-system="yes">
        <system-layout>
          <system-margins>
            <left-margin>0.00</left-margin>
            <right-margin>0.00</right-margin>
            </system-margins>
          <system-distance>150.00</system-distance>
          </system-layout>
        </print>
      <note default-x="49.08" default-y="-30.00">
        <pitch>
          <step>G</step>
          <octave>4</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        </note>
      </measure>
    <measure number="6" width="261.37">
      <note default-x="12.00" default-y="-30.00">
        <pitch>
          <step>G</step>
          <octave>4</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        </note>
      <direction placement="above">
        <direction-type>
          <words>D.S.</words>
          </direction-type>
        <sound dalsegno="segno"/>
        </direction>
      </measure>
    <measure number="7" width="261.37">
      <note default-x="12.00" default-y="-25.00">
        <pitch>
          <step>A</step>
          <octave>4</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        </note>
      </measure>
    <measure number="8" width="261.37">
      <note default-x="12.00" default-y="-30.00">
        <pitch>
          <step>G</step>
          <octave>4</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        </note>
      </measure>
    <measure number="9" width="293.37">
      <print new-system="yes">
        <system-layout>
          <system-margins>
            <left-margin>0.00</left-margin>
            <right-margin>0.00</right-margin>
            </system-margins>
          <system-distance>150.00</system-distance>
          </system-layout>
        </print>
      <note default-x="49.08" default-y="-40.00">
        <pitch>
          <step>E</step>
          <octave>4</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        </note>
      </measure>
    <measure number="10" width="261.37">
      <direction placement="above">
        <direction-type>
          <coda/>
          </direction-type>
        <sound coda="coda"/>
        </direction>
      <note default-x="12.00" default-y="-5.00">
        <pitch>
          <step>E</step>
          <octave>5</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        </note>
      </measure>
    <measure number="11" width="261.37">
      <note default-x="12.00" default-y="-15.00">
        <pitch>
          <step>C</step>
          <octave>5</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        </note>
      </measure>
    <measure number="12" width="261.37">
      <note default-x="12.00" default-y="-25.00">
        <pitch>
          <step>A</step>
          <octave>4</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        </note>
      </measure>
    <measure number="13" width="1077.49">
      <print new-system="yes">
        <system-layout>
          <system-margins>
            <left-margin>0.00</left-margin>
            <right-margin>-0.00</right-margin>
            </system-margins>
          <system-distance>150.00</system-distance>
          </system-layout>
        </print>
      <note default-x="49.08" default-y="-30.00">
        <pitch>
          <step>G</step>
          <octave>4</octave>
          </pitch>
        <duration>4</duration>
        <voice>1</voice>
        <type>whole</type>
        </note>
      <barline location="right">
        <bar-style>light-heavy</bar-style>
        </barline>
      </measure>
    </part>
  </score-partwise>
```

**Comment by craigsapp** [comment]
Perhaps `<coda>` and `<segno>` elements should be added to MEI, since these two symbols are common in music notation and are graphical (`<dir>` elements are more suitable for text).

**Comment by th-we** [comment]
Rather than introducing individual `<coda>` and `<segno>` symbols, what about a more general element called `<repeat>`, `<repeatDir>` or `<jump>`? I think all the following would fall into the realm of such an element:

* Segno/Segno symbols above the staff
* Dal Segno/D.S.
* Fine
* Da Capo/D.C.
* Da Capo al Fine/D.C. al Fine
* Da Capo al Coda
* Vamp till cue
* Maybe other non-Italian ways of saying the same thing

So these elements can have text content, symbol content or both of them combined.

Additionally, there should be logical (gestural?) domain attributes:

* `@target`: of type `@data.URI`, points to the target symbol or the measure/barline to jump to. (Or maybe rename it because we already have `@target` in [`att.pointing`](https://music-encoding.org/guidelines/v4/attribute-classes/att.pointing.html) – my understanding is that this attribute is for metadata and pointing to resources rather than elements in the music.)
* maybe something like `@nthPass` to encode on which pass the jump takes effect

As content, plain text, `<rend>`, `<symbol>` and `<lb>` are probably enough to cover all required cases. Should be of `model.controlEventLike`.

One thing that is not clear to me is what to do with the ["serpent" segnos](https://smufl-browser.edirom.de/index.html?glyphname=segnoSerpent1&glyphname=segnoSerpent2) – better encode them like `<barline glyph.name="segnoSerpent1" glyph.auth="smufl"/>`, or allow `<barline>` to have a `<repeat>` child and enforce via Schematron that it can only have a symbol, but not text?

**Comment by lpugin** [comment]
I like `<repeatDir>` because it is a good compromise between having a distinct element for each symbol, which is probably too much, and only `<dir>`, which is in my opinion too generic.

**Comment by lpugin** [comment]
Result of the discussion today: add `<coda>` and `<segno>`

**Comment by musicEnfanthen** [comment]
@lpugin connected to #1022?

**Comment by lpugin** [comment]
Yes, you are right. I think it can be closed once the PR is merged.

**Comment by musicEnfanthen** [comment]
Closed in #1022
